src/index.ts

```diff
@@ -167,6 +167,10 @@ export function presetTheme<T extends Record<string, any>>(options: PresetThemeO
               line = line.replace(/\..*?\s(.*\{)/, '$1')
             }
             return line
+          }).sort((a, b) => {
+            if (a.match(/^:root|^@media|^\}/)?.length)
+              return b.match(/^:root|^@media|^\}/)?.length ? 0 : -1
```

Comment:
Here two lines of code use the same RegExp, So we should store the RegExp to a new variable and then use it

Reply:
Ok, I'll make it clean.

Reply:
Fix it, please have a check.